### PR TITLE
Fix some more bugs with form errors with the sign up form.

### PR DESCRIPTION
**Pull request by jfly**
This stuff is crazy difficult to get right. I discovered that the birthdate field would disappear when attempting to sign up for an account and leaving certain information out. I also feature specs for the incorrect behaviors I found.

### CODE REVIEW

File: WcaOnRails/spec/features/sign_up_spec.rb [hunk 1]
```diff
@@ -82,6 +83,75 @@
 
       expect(page).to have_selector('#have-competed', visible: true)
     end
+
+    it "remembers their selected wca id on validation error" do
+      visit "/users/sign_up"
+      click_on "I have competed in a WCA competition."
+      # They have not selected a valid WCA ID yet, so don't show the birthdate verification
+      # field.
+      expect(page.find("div.user_dob_verification", visible: false).visible?).to eq false
```
Comment (jonatanklosko) [inline WcaOnRails/spec/features/sign_up_spec.rb]: Basing on a line below, couldn't you use this everywhere?
```ruby
expect(page).to have_selector("div.user_dob_verification", visible: false)
```
The current approach redundant =)

File: WcaOnRails/spec/features/sign_up_spec.rb [hunk 2]
```diff
@@ -82,6 +83,75 @@
...
+
+      selectize_input = page.find("div.user_unconfirmed_wca_id .selectize-control input")
+      selectize_input.native.send_key(person.wca_id)
+      # Wait for selectize popup to appear.
+      expect(page).to have_selector("div.selectize-dropdown", visible: true)
+      # Select item with selectize.
+      page.find("div.user_unconfirmed_wca_id input").native.send_key(:return)
```
Comment (jonatanklosko) [inline WcaOnRails/spec/features/sign_up_spec.rb]: Sorry if I'm obviously wrong, but wouldn't `selectize_input.native.send_key(:return)` do the job?

Comment (jfly) [inline WcaOnRails/spec/features/sign_up_spec.rb]: Ah, good call! I've made this change everywhere. Too much copy paste going on here =(

File: WcaOnRails/spec/features/sign_up_spec.rb (hunk 1 above)
Comment (jfly) [inline WcaOnRails/spec/features/sign_up_spec.rb]: Good point! I've cleaned up all the occurences of this in the file I could find.

**Comment by jonatanklosko** [comment]
Great, LGTM!

**Pull request merged**